src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,467 @@
+    status = GetAbsMaxCoolSetpointLimit(endpoint, &AbsMaxCoolSetpointLimit);
+
+    status = GetMinCoolSetpointLimit(endpoint, &MinCoolSetpointLimit);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        MinCoolSetpointLimit = AbsMinCoolSetpointLimit;
+    }
+
+    status = GetMaxCoolSetpointLimit(endpoint, &MaxCoolSetpointLimit);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        MaxCoolSetpointLimit = AbsMaxCoolSetpointLimit;
+    }
+
+    // Make sure the user imposed limits are within the manufacture imposed limits
+    // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3725
+    // Spec does not specify the behavior is the requested setpoint exceeds the limit allowed
+    // This implementation clamps at the limit.
+
+    // resolution of 3725 is to clamp.
+
+    if (MinCoolSetpointLimit < AbsMinCoolSetpointLimit)
+        MinCoolSetpointLimit = AbsMinCoolSetpointLimit;
+
+    if (MaxCoolSetpointLimit > AbsMaxCoolSetpointLimit)
+        MaxCoolSetpointLimit = AbsMaxCoolSetpointLimit;
+
+    if (CoolingSetpoint < MinCoolSetpointLimit)
+        CoolingSetpoint = MinCoolSetpointLimit;
+
+    if (CoolingSetpoint > MaxCoolSetpointLimit)
+        CoolingSetpoint = MaxCoolSetpointLimit;
+
+    return CoolingSetpoint;
+}
+
+bool emberAfThermostatClusterSetpointRaiseLowerCallback(EndpointId aEndpointId, chip::app::CommandHandler * commandObj,
+                                                        uint8_t mode, int8_t amount)
+{
+    int16_t HeatingSetpoint = 2000, CoolingSetpoint = 2600; // Set to defaults to be safe
+    EmberAfStatus status     = EMBER_ZCL_STATUS_FAILURE;
+    EmberAfStatus ReadStatus = EMBER_ZCL_STATUS_FAILURE;
+
+    switch (mode)
+    {
+    case EMBER_ZCL_SETPOINT_ADJUST_MODE_HEAT_AND_COOL_SETPOINTS: {
+
+        // https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3726
+        // Behavior is not specified if the device only contains one mode
+        // and the both mode command is sent.
+        // Implemented behavior is not perfect, but spec needs to be clear before changing the implementaion
+
+        // 4.3.7.2.1. Both
+        // [4.172] The client MAY indicate Both regardless of the server feature support. The server SHALL
+        // only adjust the setpoint that it supports and not respond with an error.
+
+        // Implementation assumes that the attribute will NOT be present if the device does not support that mode.
+
+        // In auto mode we will need to change both the heating and cooling setpoints
+        ReadStatus = GetOccupiedCoolingSetpoint(aEndpointId, &CoolingSetpoint);
+        if (ReadStatus == EMBER_ZCL_STATUS_SUCCESS)
+        {
+            CoolingSetpoint = static_cast<int16_t>(CoolingSetpoint + amount * 10);
+            CoolingSetpoint = EnforceCoolingSetpointLimits(CoolingSetpoint, aEndpointId);
+            status          = SetOccupiedCoolingSetpoint(aEndpointId, CoolingSetpoint);
```

Comment:
this status is lost on GetOccupiedHeadingSetpoint success

Reply:
Corrections made.  
Spec issue also written up and commented in code for auto mode if set succeeds and the other fails.